libs/application-generic/src/usecases/get-subscriber-workflow-preference/get-subscriber-workflow-preference.usecase.ts

```diff
@@ -0,0 +1,300 @@
+import { Injectable } from '@nestjs/common';
+import {
+  MessageTemplateRepository,
+  NotificationTemplateEntity,
+  SubscriberEntity,
+  SubscriberPreferenceRepository,
+  SubscriberRepository,
+  TenantRepository,
+  WorkflowOverrideRepository,
+} from '@novu/dal';
+import {
+  ChannelTypeEnum,
+  IOverridePreferencesSources,
+  IPreferenceChannels,
+  IPreferenceOverride,
+  ISubscriberWorkflowPreferenceResponse,
+  ITemplateConfiguration,
+  PreferenceLevelEnum,
+  PreferenceOverrideSourceEnum,
+  StepTypeEnum,
+} from '@novu/shared';
+
+import { GetSubscriberWorkflowPreferenceCommand } from './get-subscriber-workflow-preference.command';
+
+import { buildSubscriberKey, CachedEntity } from '../../services/cache';
+import { ApiException } from '../../utils/exceptions';
+
+const PRIORITY_ORDER = [
+  PreferenceOverrideSourceEnum.TEMPLATE,
+  PreferenceOverrideSourceEnum.WORKFLOW_OVERRIDE,
+  PreferenceOverrideSourceEnum.SUBSCRIBER,
+];
+
+@Injectable()
+export class GetSubscriberWorkflowPreference {
```

Comment:
☑ **chore:** I would suggest not duplicate this code but rather creating a `mapper` function that will change the `template` to `workflow`